FreeRTOS-Plus/Demo/coreSNTP_Windows_Simulator/DemoTasks/SNTPExample.c

```diff
@@ -0,0 +1,772 @@
+ *
+ * https://www.FreeRTOS.org
+ * https://github.com/FreeRTOS
+ *
+ */
+
+/*
+ * Demo for showing use of the coreSNTP library for synchronizing system time
+ * with the internet and maintaining correct wall-clock time.
+ *
+ * The example shown below shows how an SNTP client task can be created using
+ * the coreSNTP library to periodically synchronize system clock with SNTP/NTP
+ * servers, and how an separate task (like an application task) query current
+ * time from the system.
+ *
+ * !!! NOTE !!!
+ * This SNTP demo does not authenticate the server nor the client.
+ * Hence, this demo should not be used as production ready code.
+ */
+
+/* Standard includes. */
+#include <string.h>
+#include <stdio.h>
+#include <stdint.h>
+#include <time.h>
+
+/* Kernel includes. */
+#include "FreeRTOS.h"
+#include "task.h"
+
+/* Demo Specific configs. */
+#include "demo_config.h"
+
+/* SNTP library includes. */
+#include "core_sntp_client.h"
+
+/* Synchronization primitive include. */
+#include "semphr.h"
+
+/* FreeRTOS+TCP includes */
+#include "FreeRTOS_IP.h"
+#include "FreeRTOS_UDP_IP.h"
+#include "FreeRTOS_Sockets.h"
+
+/*-----------------------------------------------------------*/
+
+/* Compile time error for undefined configs. */
+
+#ifndef democonfigLIST_OF_TIME_SERVERS
+    #error "Define the democonfigLIST_OF_TIME_SERVERS config by following the instructions in demo_config.h file."
+#endif
+
+#ifndef democonfigDESIRED_CLOCK_ACCURACY_MS
+    #error "Define the democonfigDESIRED_CLOCK_ACCURACY_MS config by following instructions in demo_config.h file."
+#endif
+
+#ifndef democonfigSYSTEM_CLOCK_TOLERANCE_PPM
+    #error "Define the democonfigSYSTEM_CLOCK_TOLERANCE_PPM config by following instructions in demo_config.h file."
+#endif
+
+#ifndef democonfigSYSTEM_START_YEAR
+    #error "Define the democonfigSYSTEM_START_YEAR config by following instructions in demo_config.h file."
+#endif
+
+/*-----------------------------------------------------------*/
+/* Default values for configuration . */
+
+#ifndef democonfigSERVER_RESPONSE_TIMEOUT_MS
+    #define democonfigSERVER_RESPONSE_TIMEOUT_MS    ( 5000 )
+#endif
+
+/**
+ * @brief The size for network buffer that is allocated for initializing the coreSNTP library in the
+ * demo.
+ *
+ * @note The size of the buffer MUST be large enough to hold an entire SNTP packet, which includes the standard SNTP
+ * packet data of 48 bytes and authentication data for security mechanism, if used, in communication with time server.
+ */
+#define SNTP_CONTEXT_NETWORK_BUFFER_SIZE    ( SNTP_PACKET_BASE_SIZE )
+
+/**
+ * @brief The periodicity of the application task in query and logging system time. This is the time
+ * interval between consecutive system clock time queries in the sample application task.
+ */
+#define CLOCK_QUERY_TASK_DELAY_MS           ( 1000 )
+
+/**
+ * @brief The constant for storing the number of milliseconds per FreeRTOS tick in the system.
```

Comment:
This is inaccurate on Winsim, right? Maybe good to mention that if so

Reply:
I will add a note saying that WinSim has a clock drift and this represents the number of milliseconds according to FreeRTOS system.